### ISSUE TITLE
Unable to restore channel backup on new device

### ISSUE THREAD

**Issue by Horndev** [btcontract/lnwallet#28]
Old device: Android 8.1.0
New device: Android 10

I was following the directions here: https://lightning-wallet.com/migrations#migrations

I place the backup file in `/BLW/mainnet` folder.

- I start BLW
- select Restore Wallet
- grant permission for file access when requested.
- I enter my BIP32 phrase

I then get the following error when trying to restore the wallet:

```
Channel backup file is found but could not be decrypted: 
<</storage/emulated/0/BLW/mainnet/encrypted.channels.bkp-...: open failed: EACCESS (Permission denied)>>
```

It appears that the location is correct as BLW does find the backup file, however, the error message suggests that the app is being denied permission to open it.

**Comment by btcontract** [comment]
Does an app have a permission to access a file system? (this can be checked in system settings).

**Comment by Horndev** [comment]
Yes it does.  As I mentioned in step 3, I granted permission.  I verified that it does have permission to access file system.  I also granted permission to access camera.

**Comment by btcontract** [comment]
Is it possible to try the same on some other device? FWIW just recently there was an issue which looked very similarly: user had `EACCESS (Permission denied)` on one phone but not on the other.

A good solution to this would be to reverse the whole procedure: a backup file should first be located in file system (or messaging app) and then shared with fresh BLW, without making BLW to scan a system.

**Comment by Horndev** [comment]
I do not have another device to try this on, unfortunately.  Could you please elaborate on your proposed solution?

How do I do the step "a backup file should first be located in file system".

**Comment by btcontract** [comment]
It's not possible currently, I've just made a note about how to re-implement this properly in the future. Meanwhile I don't think I can help anyhow since I don't have an access to device or an app.

**Comment by Horndev** [comment]
That would be a very nice feature.  Being able to have good backups of channels is essential.
